cylc/flow/task_events_mgr.py

```diff
@@ -682,75 +692,80 @@ def process_message(
         # Satisfy my output, if possible, and spawn children.
         # (first remove signal: failed/EXIT -> failed)
 
+        # Complete the corresponding task output, if there is one.
         msg0 = message.split('/')[0]
-        completed_trigger = itask.state.outputs.set_msg_trg_completion(
-            message=msg0, is_completed=True)
-        self.data_store_mgr.delta_task_output(itask, msg0)
-
-        # Check the `started` event has not been missed e.g. due to
-        # polling delay
-        if (message not in [self.EVENT_SUBMITTED, self.EVENT_SUBMIT_FAILED,
-                            self.EVENT_STARTED]
-                and not itask.state.outputs.is_completed(TASK_OUTPUT_STARTED)):
-            self.setup_event_handlers(
-                itask, self.EVENT_STARTED, f'job {self.EVENT_STARTED}')
-            self.spawn_func(itask, TASK_OUTPUT_STARTED)
+        if message.startswith(ABORT_MESSAGE_PREFIX):
+            msg0 = TASK_OUTPUT_FAILED
+
+        completed_output = None
+        if msg0 not in [TASK_OUTPUT_SUBMIT_FAILED, TASK_OUTPUT_FAILED]:
+            completed_output = itask.state.outputs.set_msg_trg_completion(
+                message=msg0, is_completed=True)
+            if completed_output:
+                self.data_store_mgr.delta_task_output(itask, msg0)
```

Comment:
@hjoliver Can you explain why the  `if msg0 not in [TASK_OUTPUT_SUBMIT_FAILED, TASK_OUTPUT_FAILED]`?

Reply:
Ah, I don't recall sorry. I suppose (guessing) the output was completed elsewhere in those cases.  Have you found a bug associated with this?

Reply:
Not found a bug but trying to understand. Ah, I think this is because the task should be in the submit-failed or failed state if there a retry delays still pending